packages/react/src/components/Popup/Popup.tsx

```diff
@@ -212,10 +218,23 @@ export default class Popup extends AutoControlledComponent<PopupProps, PopupStat
     rtl,
     accessibility,
   }: RenderResultConfig<PopupProps>): React.ReactNode {
-    const { inline, mountNode } = this.props
+    const { inline, mountNode, trapFocus, autoFocus } = this.props
     const { open } = this.state
     const popupContent = open && this.renderPopupContent(classes.popup, rtl, accessibility)
 
+    if (process.env.NODE_ENV !== 'production') {
+      if (inline && trapFocus) {
+        console.warn(
+          'Using "trapFocus" in inline popup leads to broken behavior for screen reader users.',
+        )
+      }
+      if (!inline && autoFocus) {
+        console.warn(
+          'Beware, "autoFocus" prop will just grab focus at the moment of mount and will not trap it. As user is able to TAB out from popup, better use "inline" prop to keep correct tab order.',
+        )
+      }
+    }
```

Comment:
Not sure about logging in `render()`, it will produce a warning on each render cycle.

What about `componentDidMount()`?

Reply:
but then if user updates some of the props, component re-renders, the console warn will disappear even if the problem is not fixed. Isn't it? So maybe componentDidUpdate?

Reply:
I am not that it's a real scenario. If it's really breaks accessibility probably it should throw an error instead. Let's discuss it offline